src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,146 @@
+package duke.ui;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Task;
+
+/**
+ * The UI component for the fanTASKtic.
+ * Handles interactions with the user, such as reading commands and displaying messages.
+ */
+public class Ui {
+
+    private static final String LOGO = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+
+    private Scanner sc;
+
+    /**
+     * Constructs a Ui object and initializes the Scanner for user input.
+     */
+    public Ui() {
+        this.sc = new Scanner(System.in);
+    }
+
+    /**
+     * Reads a command from the user.
+     *
+     * @return The user's input command.
+     */
+    public String readCommand() {
+        return sc.nextLine();
+    }
+
+    /**
+     * Prints the welcome message when the chatbot starts.
+     */
+    public void welcomeMsg() {
```

Comment:
Method names should be verbs. Same goes for `goodbyeMsg` below
```suggestion
    public void showWelcomeMsg() {
```